<launch>
    <arg name="subscribe_topic" default="/hesai/pandar" />

    <arg name="vertical_removal" default="true" />
    <arg name="below_distance" default="-1.3" />
    <arg name="above_distance" default="0.5" />

    <arg name="lateral_removal" default="false" />
    <arg name="left_distance" default="3" />
    <arg name="right_distance" default="3" />

    <node pkg="space_filter" type="space_filter" name="space_filter">
        <param name="subscribe_topic" value="$(arg subscribe_topic)" />
        <param name="vertical_removal" value="$(arg vertical_removal)" />
        <param name="below_distance" value="$(arg below_distance)" />
        <param name="above_distance" value="$(arg above_distance)" />

        <param name="lateral_removal" value="$(arg lateral_removal)" />
        <param name="left_distance" value="$(arg left_distance)" />
        <param name="right_distance" value="$(arg right_distance)" />
    </node>

    <node pkg="rviz" type="rviz" name="rviz" />
</launch>